client/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+#![doc = include_str!("../README.md")]
+
+mod blob;
+mod blobstream;
+mod client;
+mod fraud;
+mod header;
+mod share;
+mod state;
+mod utils;
+
+/// API related types.
+pub mod api {
+    pub use crate::blob::BlobApi;
+    pub use crate::blobstream::BlobstreamApi;
+    pub use crate::fraud::FraudApi;
+    pub use crate::header::HeaderApi;
+    pub use crate::share::ShareApi;
+    pub use crate::state::StateApi;
+
+    /// Blob API related types.
+    pub mod blob {
+        #[doc(inline)]
+        pub use crate::blob::BlobsAtHeight;
+    }
+
+    /// Share API related types.
+    pub mod share {
+        #[doc(inline)]
+        pub use crate::share::{GetRangeResponse, GetRowResponse, RowSide, SampleCoordinates};
+    }
+}
+
+/// TX related types.
+pub mod tx {
+    #[doc(inline)]
+    pub use celestia_grpc::{DocSigner, IntoProtobufAny, SignDoc, TxConfig, TxInfo};
+}
+
+pub use crate::client::{Client, ClientBuilder};
+
+/// Alias for a `Result` with the error type [`celestia_client::Error`].
+///
+/// [`celestia_client::Error`]: crate::Error
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Representation of all the errors that can occur when interacting with [`celestia_client`].
+///
+/// [`celestia_client`]: crate
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// Celestia RPC error.
+    #[error("RPC error: {0}")]
+    Rpc(#[from] celestia_rpc::Error),
+
+    /// Celestia GRPC error.
+    #[error("GRPC error: {0}")]
+    Grpc(#[from] celestia_grpc::Error),
+
+    /// Celestia types error.
+    #[error("Celestia types error: {0}")]
+    Types(#[from] celestia_types::Error),
+
+    /// Client is in read-only mode.
+    #[error("Client is constructed for read-only mode")]
```

Comment:
```suggestion
    #[error("Client is constructed for read-only mode, operation not supported")]
```
maybe more context here?